public/app/features/explore/utils/links.ts

```diff
@@ -20,6 +20,10 @@ export function getFieldLinksForExplore(
   const data = getLinksFromLogsField(field, rowIndex);
   return data.map(d => {
     if (d.link.meta?.datasourceUid) {
+      // TODO: fix the ambiguity here
+      // This looks weird but in case meta.datasourceUid is set we save the query in url which will get
+      // interpolated into href
+      const parsedQuery = JSON.parse(d.linkModel.href);
```

Comment:
Problem 1. Probably could do this in a type safe way (at least a bit more typesafe) though not sure where such changes would ripple.